diesel_compile_tests/tests/compile-fail/pg_specific_expressions_cant_be_used_in_a_sqlite_query.rs

```diff
@@ -1,38 +0,0 @@
-#[macro_use] extern crate diesel;
-
-use diesel::*;
-use diesel::sql_types::*;
-use diesel::dsl::*;
-
-table! {
-    users {
-        id -> Integer,
-        name -> VarChar,
-    }
-}
-
-sql_function!(fn lower(x: VarChar) -> VarChar);
-
-#[derive(Insertable)]
-#[table_name="users"]
-struct NewUser(#[column_name = "name"] &'static str);
-
-// NOTE: This test is meant to be comprehensive, but not exhaustive.
-fn main() {
-    use self::users::dsl::*;
-    let connection = SqliteConnection::establish(":memory:").unwrap();
-
-    users.select(id).filter(name.eq(any(Vec::<String>::new())))
-        .load::<i32>(&connection);
-    //~^ ERROR type mismatch resolving `<diesel::SqliteConnection as diesel::Connection>::Backend == diesel::pg::Pg`
-    users.select(id).filter(name.is_not_distinct_from("Sean"))
-        .load::<i32>(&connection);
-    //~^ ERROR type mismatch resolving `<diesel::SqliteConnection as diesel::Connection>::Backend == diesel::pg::Pg`
-    users.select(id).filter(now.eq(now.at_time_zone("UTC")))
-        .load::<i32>(&connection);
-    //~^ ERROR type mismatch resolving `<diesel::SqliteConnection as diesel::Connection>::Backend == diesel::pg::Pg`
-    insert_into(users).values(&NewUser("Sean"))
```

Comment:
Only remove this statement all other parts of the test should not be changed.

Reply:
After looking at the code above I think this should be changed to contain a `on_constrain` clause, so verify that this extension is only supported on postgresql (Sorry for changing my mind so often…)